### ISSUE TITLE
Ability to try webhook with fake events

### ISSUE THREAD

**Issue by JoseRenan** [opencollective/opencollective#4071]
## User story 

As a collective admin, I would like to emit fake webhooks events to check if they are configured correctly.

## Best solution for this problem

On Gitlab, you can test webhooks as shown above, selecting an event and the webhook will be emitted with a test message.

![image](https://user-images.githubusercontent.com/11728746/110543549-3b07fc80-8109-11eb-8b1b-39b7365967bc.png)

**Comment by Betree** [comment]
I really think this would be the best way to document the webhook events, and that it's not too complex to implement. My recommendation would be to update `server/lib/webhooks.js`:

1. Define payload fields in a declarative way (*example below must be adapted to cover all use cases*):

```es6
/**
 * Usage: getSubFields('parent', ['child1', 'child2'])
 * Returns: ['parent.child1', 'parent.child2']
 */
const getSubFields = (parent, fields) => {
  return fields.map(field => `${parent}.${field}`);
};

const ACTIVITY_DATA_FIELDS = {
  [activities.COLLECTIVE_TRANSACTION_CREATED]: getSubFields('transaction', ['amount', 'currency']),
  [activities.COLLECTIVE_UPDATE_PUBLISHED]: getSubFields('update', ['html', 'title', 'slug', 'tags', 'isPrivate']),
  [activities.COLLECTIVE_MEMBER_CREATED]: getSubFields('member', ['role', 'description', 'since']),
};
```

2. Once get this object, we'll be able to do something like:

```es6
export const generateFakeWebhookPayload = (type) => {
  const definition = ACTIVITY_DATA_FIELDS[type];
  return mapValues(definition, (field) => {
    switch (field.type) {
      case 'collective':
        return fakeCollective;
      case 'tier':
        return fakeTier;
      case 'order':
        return fakeOrder;
    }
  }))
}
```

3. We plug this on a `sendTestWebhookEvent` (or similar) mutation

**Comment by SamWhited** [comment]
> I really think this would be the best way to document the webhook events

This is definitely a nice feature to have, and would make development *significantly* easier, but I wouldn't call it a replacement for good documentation. I would call these two separate issues that both need to be fixed, personally. It might be that the docs provide a high-level overview and the test events are considered more up-to-date and the canonical way to know what specific fields are sent. But having a way to read about what events to support *before* you do an implementation (which would be required to receive test events) would be extremely helpful too.

**Comment by Betree** [comment]
> > I really think this would be the best way to document the webhook events
> 
> This is definitely a nice feature to have, and would make development _significantly_ easier, but I wouldn't call it a replacement for good documentation. I would call these two separate issues that both need to be fixed eventually if and when they can be, personally. It might be that the docs provide a high-level overview and the test events are considered more up-to-date and the canonical way to know what specific fields are sent. But having a way to read about what events to support _before_ you do an implementation (which would be required to receive test events) would be extremely helpful too.

You're right that it does not replace docs, but maintaining good and comprehensive documentation for all webhook events is hard and it will be difficult to achieve given our current bandwidth and priorities. Fake events on the other hand can easily be hacked in a small project.

**Comment by SamWhited** [comment]
Understood; that is why I would suggest that they be separate issues. Hopefully this is low-hanging fruit, and having docs is something that can only happen if/when the team has the bandwidth (which may be never, of course).

**Comment by adambirds** [comment]
I would be happy with this as an alternative to #4244 which I asked for. Doesn't take much to get them sent to https://webhook.site or similar. I have done this most. once this has sent fake webhook events it would also be easy to add them to documentation too.

**Comment by Betree** [comment]
Related update: https://github.com/opencollective/opencollective/issues/4244#issuecomment-1187064424
